scripts/utils/Git.ts

```diff
@@ -0,0 +1,261 @@
+
+                    currentHunk = {
+                        oldStart,
+                        oldCount,
+                        newStart,
+                        newCount,
+                        lines: [],
+                    };
+                }
+                continue;
+            }
+
+            // Diff content lines
+            if (currentHunk && currentFile && line.length > 0) {
+                const firstChar = line[0];
+                const content = line.slice(1); // Remove the +/- prefix
+
+                if (firstChar === '+') {
+                    // For added lines, use new file line numbers
+                    const lineNumber = this.calculateLineNumber(currentHunk, 'added');
+
+                    currentHunk.lines.push({
+                        lineNumber,
+                        type: 'added',
+                        content,
+                    });
+                    currentFile.addedLines.add(lineNumber);
+                } else if (firstChar === '-') {
+                    // For removed lines, use old file line numbers
+                    const lineNumber = this.calculateLineNumber(currentHunk, 'removed');
+
+                    currentHunk.lines.push({
+                        lineNumber,
+                        type: 'removed',
+                        content,
+                    });
+                    currentFile.removedLines.add(lineNumber);
+                } else if (firstChar === ' ') {
+                    // For context lines, use new file line numbers
+                    const lineNumber = this.calculateLineNumber(currentHunk, 'context');
+
+                    currentHunk.lines.push({
+                        lineNumber,
+                        type: 'context',
+                        content,
+                    });
+                }
+            }
+        }
+
+        // Add the last file and hunk
+        if (currentHunk && currentFile) {
+            currentFile.hunks.push(currentHunk);
+        }
+        if (currentFile) {
+            files.push(currentFile);
+        }
+
+        // Calculate modified lines (lines that have both additions and removals)
+        for (const file of files) {
+            for (const hunk of file.hunks) {
+                const addedInHunk = new Set<number>();
+                const removedInHunk = new Set<number>();
+
+                for (const line of hunk.lines) {
+                    if (line.type === 'added') {
+                        addedInHunk.add(line.lineNumber);
+                    } else if (line.type === 'removed') {
+                        removedInHunk.add(line.lineNumber);
+                    }
+                }
+
+                // Lines that are both added and removed (in close proximity) are considered modified
+                for (const addedLine of addedInHunk) {
+                    for (const removedLine of removedInHunk) {
+                        if (Math.abs(addedLine - removedLine) <= 2) {
```

Comment:
Hmm why =2? Shouldn't a removed modified line be the very next one?

Reply:
chatted on a call but I think you're right that this doesn't make sense